obs-studio-client/source/controller.cpp

```diff
@@ -24,13 +24,48 @@
 #include <iostream>
 #include <nan.h>
 #include <fstream>
+#include <locale>
+#include <codecvt>
 
 #pragma region Windows
 #ifdef _WIN32
 #include <windows.h>
 #include <direct.h>
 #include <wchar.h>
 
+/* Length is the number of characters, not bytes */
+namespace {
+
+std::string from_utf16_wide_to_utf8(const wchar_t *from, size_t length = -1)
```

Comment:
It looks like we might need these more often, perhaps put them in utility.hpp/cpp so they can be re-used?

Reply:
That's not a bad idea. If we use them often, I should also cache those conversion objects as they can be horrifically slow to create.